### PR TITLE
feat: open status text user profile links on local server

### PR DESCRIPTION
**Pull request by llun**
## Summary
Rewrites user profile links (mentions, actor profiles, Bluesky bridge handles) embedded in status text and notifications to navigate to local profile routes (`/@username` or `/@username@domain`) instead of opening the remote server in a new browser window.

- Added `lib/utils/text/extractProfileHref.ts` to parse and resolve profile URLs and status tags against instance host.
- Updated `lib/utils/text/cleanClassName.ts` to replace matching `<a>` tags with Next.js `<Link>` pointing to local routes, removing `target="_blank"` and setting `prefetch={false}` / `onClick={(e) => e.stopPropagation()}`.
- Updated `Post`, `StatusNotification`, and `ActivityImportNotification` to pass `host` and `tags` into `cleanClassName`.

## Verification
- All local pre-commit gates pass in order:
  - `yarn run prettier --write .`
  - `yarn lint` (0 errors)
  - `yarn typecheck` (0 errors)
  - `yarn build` (success)
  - `yarn test` (1053 test files, 14,584 tests all passing)
- Added comprehensive unit tests for `extractProfileHref`, `cleanClassName`, and `Post` rendering.
- `package.json` version untouched.

### CODE REVIEW

File: lib/utils/text/extractProfileHref.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    return null
+  }
+}
+
+export const extractProfileHref = (
+  href: string | undefined,
+  options?: ExtractProfileHrefOptions
+): string | undefined => {
+  if (!href) return undefined
+
```
Comment (llun) [inline lib/utils/text/extractProfileHref.ts:81]: Explicitly guard protocol against non-HTTP(S) schemes (e.g. `javascript:`, `data:`) to prevent unexpected URL resolution.

File: lib/utils/text/extractProfileHref.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  if (href.startsWith('/@')) {
+    return href
+  }
+
+  // 2. Resolve via status mention tags when available (covers non-standard fediverse profile URLs).
+  if (options?.tags && options.tags.length > 0) {
+    const matchingTag = options.tags.find(
+      (tag) => tag.type === 'mention' && tag.value === href
+    )
+    if (matchingTag) {
```
Comment (llun) [inline lib/utils/text/extractProfileHref.ts:113]: Normalize tag mention URLs (e.g. trailing slashes) and host comparison (e.g. host with or without ports in dev environments).

File: lib/components/posts/post.test.tsx [hunk 3]
```diff
@@ -2467,4 +2467,44 @@ describe('Post', () => {
...
+          status={statusWithMention}
+          onShowAttachment={vi.fn()}
+        />
+      )
+
+      const mentionLink = screen.getByRole('link', { name: '@remoteuser' })
+      expect(mentionLink).toHaveAttribute(
+        'href',
+        '/@remoteuser@mastodon.social'
+      )
```
Comment (llun) [inline lib/components/posts/post.test.tsx:2509]: Add assertion or test verifying that clicking the rewritten profile link stops click propagation.

**Review by llun — COMMENTED** [review]
### Round 1 Code Review
Identified findings regarding protocol validation, URL/host normalization, and test assertions for click propagation.

File: lib/utils/text/extractProfileHref.ts (hunk 1 above)
Comment (llun) [reply lib/utils/text/extractProfileHref.ts:81]: Addressed in commit e491be291: added explicit URL protocol guard requiring http: or https: for non-relative profile URLs, and added test cases verifying rejection of javascript:, data:, and file: schemes.

File: lib/utils/text/extractProfileHref.ts (hunk 2 above)
Comment (llun) [reply lib/utils/text/extractProfileHref.ts:113]: Addressed in commit e491be291: added isHostMatch helper supporting port normalization between local host and URL domains, and added normalizeUrlForMatch to handle trailing slash variations in mention tag values.

File: lib/components/posts/post.test.tsx (hunk 3 above)
Comment (llun) [reply lib/components/posts/post.test.tsx:2509]: Addressed in commit e491be291: added assertion in post.test.tsx firing a click event on the mention link inside a parent click listener and verifying event propagation was stopped.

**Review by llun — COMMENTED** [review]
### Round 2 Code Review (Clean)
All Round 1 findings have been resolved and verified. No actionable issues identified. The diff is clean and adheres to repository guidelines.

File: lib/utils/text/extractProfileHref.ts [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+}
+
+export const extractProfileHref = (
+  href: string | undefined,
+  options?: ExtractProfileHrefOptions
+): string | undefined => {
+  if (!href) return undefined
+
+  // 1. Root-relative handle paths are already local profile destinations.
+  if (href.startsWith('/@')) {
```
Comment (llun) [inline lib/utils/text/extractProfileHref.ts]: Strictly validate root-relative profile destinations: \`href.startsWith(\"/@\")\` permits status permalinks, subpaths, and malformed/open-redirect paths like \`/@//evil.com\`.

File: lib/utils/text/extractProfileHref.ts [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+  }
+}
+
+const getProfileHandleFromParts = (parts: string[]) => {
+  const profileIndex = parts.indexOf('profile')
+  const profileHandle = parts[profileIndex + 1]
+  if (
+    profileIndex >= 0 &&
+    profileHandle &&
+    !isOpaqueActorUsernameValue(profileHandle)
```
Comment (llun) [inline lib/utils/text/extractProfileHref.ts]: Check that Bluesky URL path has exactly 2 segments (\`[\"profile\", handle]\`) so post, feed, and list URLs are not falsely rewritten as profile links.

File: lib/utils/text/extractProfileHref.ts [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+        (tag.value === href ||
+          normalizeUrlForMatch(tag.value) === normalizedTarget)
+    )
+    if (matchingTag) {
+      const cleanName = matchingTag.name.replace(/^@+/, '')
+      const atIndex = cleanName.indexOf('@')
+      let username: string
+      let domain: string
+      if (atIndex > 0) {
+        username = cleanName.slice(0, atIndex)
```
Comment (llun) [inline lib/utils/text/extractProfileHref.ts]: Validate that mention tag names do not contain multiple \`@\` characters.

**Review by llun — COMMENTED** [review]
### Code Review Findings (Round 2)
Detailed review surfaced edge cases in root-relative path validation, Bluesky post URL handling, consistency across status-rendering surfaces, and test assertions.

File: lib/utils/text/extractProfileHref.ts (hunk 4 above)
Comment (llun) [reply lib/utils/text/extractProfileHref.ts]: Addressed in commit 38b0392de: strictly validate root-relative handle paths to match /@username or /@username@domain without subpaths, rejecting status permalinks, subpaths, and malformed/open-redirect paths like /@//evil.com.

File: lib/utils/text/extractProfileHref.ts (hunk 5 above)
Comment (llun) [reply lib/utils/text/extractProfileHref.ts]: Addressed in commit 38b0392de: enforced profileIndex === 0 && parts.length === 2 so post, feed, and list URLs are not falsely rewritten as profile links.

File: lib/utils/text/extractProfileHref.ts (hunk 6 above)
Comment (llun) [reply lib/utils/text/extractProfileHref.ts]: Addressed in commit 38b0392de: validated that cleanName.split('@').length <= 2 to safely ignore malformed mention tags with multiple @ symbols.

**Review by llun — COMMENTED** [review]
### Round 3 Code Review (Clean)
All Round 2 findings have been verified and resolved. No actionable issues identified. The diff is clean.

**Comment by llun** [comment]
Commit `38c1a18a9` refined `isHostMatch` cross-port comparison (ensuring distinct explicit ports on the same host do not match) and hardened mention tag shape validation. All DoD gates pass.
